README.md

```diff
@@ -172,34 +194,137 @@ On iOS installation is a bit more complicated, because you need to setup a scree
+   $ cd example
+   $ yarn
+   ```
+4. In App.ts replace server url with your server's url.
+5. `yarn run android` or `yarn run ios` or run project from Android Studio / Xcode just like every RN project. Note that simulators won't work, you have to test on real device for camera and screensharing to run.
+
+# Usage
 
-## Usage
+Important note!! Since version 7.4.0 call function `initializeWebRTC()` once in your app before using any other functionality.
 
-Since version 0.2.0 call function initializeWebRTC() once in your app before using any other functionality.
+Start with connecting to the membrane webrtc server. Use `useWebRTC()` hook to manage connection:
 
-Make sure you have:
+```ts
+const { connect, disconnect, error } = useWebRTC();
+```
+
+Connect to the server and join the room using the `connect` function. Use user metadata to pass things like usernames etc. to the server. You can also pass connection params that will be sent to the socket when establishing the connection tries.
+
+```ts
+const startServerConnection = () => {
+  try {
+    await connect('https://example.com', "Annie's room", {
+      endpointMetadata: {
+        displayName: 'Annie',
+      },
+      connectionParams: {
+        token: 'TOKEN',
+      },
+    });
+  } catch (e) {
+    console.log('error!');
+  }
+};
+```
 
-- Running [Jellyfish](https://github.com/jellyfish-dev/jellyfish) server.
-- Created room and token of peer in that room.
-  You can use [dashboard](https://github.com/jellyfish-dev/jellyfish-react-client/tree/main/examples/dashboard) example to create room and peer token.
+Remember to gracefully disconnect from the server using the `disconnect()` function:
 
-You can refer to our minimal example on how to use this library.
+```ts
+const stopServerConnection = () => {
+  await disconnect();
+};
+```
+
+Also handle errors properly, for example when internet connection fails or server is down:
+
+```ts
+useEffect(() => {
+  if (error) console.log('error: ', e);
+}, [error]);
+```
+
+Start the device's camera and microphone using `useCamera()` and `useMicrophone()` hooks. Use `videoTrackMetadata` and `audioTrackMetadata` options to send metadata about the tracks (for example whether it's a camera or screencast track).
+
+```ts
+const { startCamera } = useCamera();
+const { startMicrophone } = useMicrophone();
+
+await startCamera({
+  quality: VideoQuality.HD_169,
+  videoTrackMetadata: { active: true, type: 'camera' },
+});
+await startMicrophone({ audioTrackMetadata: { active: true, type: 'audio' } });
+```
+
+For more options and functions to control the camera and microphone see the API documentation.
+
+If you have the connection set up, then use `useEndpoints()` hook to track the other endpoints in the room. One of the endpoints will be a local participant (the one who's using the device). When endpoints is added or removed because an user joins or leaves the room, the endpoints will be updated automatically. Simply call the hook like this:
```

Comment:
Please set up line wrapping (eg. [like this](https://github.com/jellyfish-dev/js-server-sdk/blob/137dcd4196a36216041855f8a6bc66ca7bbf4807/.prettierrc#L4)) and please make sure prettier will format `.md` files

Reply:
Added as follow up task